conans/test/integration/settings/remove_subsetting_test.py

```diff
@@ -144,3 +145,60 @@ def build(self):
         client.run("package .")
         self.assertIn("ERROR: PACKAGE 'settings.compiler.libcxx' doesn't exist for 'gcc'",
                       client.out)
+
+
+def test_settings_and_options_rm_safe():
+    client = TestClient()
+    conanfile = textwrap.dedent("""
+    from conan import ConanFile
+    class Pkg(ConanFile):
+        settings = "os", "build_type", "compiler"
+        options = {"opt1": [True, False], "opt2": [True, False]}
+        default_options = "opt1=True", "opt2=False"
+
+        def configure(self):
+            # setting
+            self.settings.rm_safe("build_type")
+            # sub-setting
+            self.settings.rm_safe("compiler.version")
+            # wrong settings
+            self.settings.rm_safe("fake_field")
+            self.settings.rm_safe("fake_field.version")
+
+        def config_options(self):
+            # option
+            self.options.rm_safe("opt2")
+            # wrong option
+            self.options.rm_safe("opt15")
+
+        def source(self):
+            try:
+                self.settings.build_type
```

Comment:
accessing settings in ``source`` is discouraged, even blocked in 2.0. Maybe move this checks to ``build()`` too, or to ``generate()``